### PR TITLE
Added special mention feature

### LINKED ISSUE #612: Add support for special mentions custom extension
https://www.loom.com/share/2320c7010e844ba686391e737b53e333

@gaagul _a

### PR DESCRIPTION
**Pull request by gaagul**
Fixes #612 

**Description**
Added: support for special mentions

**Checklist**

- [x] I have added the necessary label (patch/minor/major - If package publish is required).

**Reviewers**
@AbhayVAshokan _a

### CODE REVIEW

**Comment by gaagul** [comment]
@AbhayVAshokan Right now the special-mentions extension is using `MentionList` and `utils` similar to mention extension,
I have written it separately considering any future customization requirements, Would you like me to import from the same file or keep separate files for both extensions? 
Please let me know your thoughts.

**Comment by AbhayVAshokan** [comment]
Let's keep them separate. This will provide a consistency to the folder structure and naming convention (`ExtensionName/ExtensionConfig.js`).

File: lib/components/Editor/CustomExtensions/useCustomExtensions.js [hunk 1]
```diff
@@ -85,6 +86,14 @@ const useCustomExtensions = ({
         },
       })
     );
+
+    customExtensions.push(
+      SpecialMentions.configure({
+        suggestion: {
+          items: createMentionSuggestions(mentions),
```
Comment (AbhayVAshokan) [inline lib/components/Editor/CustomExtensions/useCustomExtensions.js:93]: Add the following line in line 82.
```jsx
const items =  createMentionSuggestions(mentions);
```

Pass it to both the `configure` methods.
```suggestion
          items
```

**Review by AbhayVAshokan — CHANGES_REQUESTED** [review]
@gaagul _a

The duplicate files need to be deleted. These are exactly similar to `Mentions`. I assumed that there will be some changes in the previous comment.

- [x] Update documentation.
- [x] Both mentions and special mentions should have similar styling.
image

**Comment by gaagul** [comment]
@AbhayVAshokan _a I have optimized the code and also removed the duplicate files, I have also removed the index file since we are not adding a button to the menu for special-mentions, and also cleared unused command from the editor because it is only used as part of the menu which in this case is not needed.

Please review the changes.

**Review by AbhayVAshokan — APPROVED** [review]
Nicely done 👌

**Pull request merged**